src/uct/api/uct.h

```diff
@@ -1098,6 +1111,25 @@ struct uct_cm_attr {
 };
 
 
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief UCT listener attributes, capabilities and limitations.
+ */
+struct uct_listener_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_listener_attr_field. Fields not specified by this mask
+     * will be ignored.
+     */
+    uint64_t        field_mask;
+
+    /**
+     * Sockaddr on which this listener is listening.
+     */
+    ucs_sock_addr_t sockaddr;
```

Comment:
`ucs_sock_addr_t` has a pointer to `const struct sockaddr   *addr`, who is owner of this memory? should it be freed or how long it's accessible?

Reply:
it should be valid as long as the listener is valid.

Reply:
should we document this?

Reply:
i think it makes more sense to define it as struct sockaddr_storage rather than pointer to sockaddr